axolotls/numeric_column.py

```diff
@@ -133,12 +133,29 @@ def presence(self) -> Optional[torch.BoolTensor]:
         return self._presence
 
     def __len__(self) -> int:
-        return len(self._values)
+        return self._values.numel()
 
     def to_arrow(self):
         # TODO: Check whether PyArrow is available 
         import pyarrow as pa
 
+        if not self.presence is not None and self.values.dtype != torch.bool:
```

Comment:
Emmm. I am a little bit confused by the condition here. Should it be
`if self.presence is not None and self.values.dtype != torch.bool:`?

Reply:
ah, my bad, it should be 
```
    if self.presence is None and self.values.dtype != torch.bool:
```

Basically, we can do direct buffer conversion if both satisifes:
* The `NumericColumn` is not nullable 
* `NumericColumn` doesn't store bool dtype data. (I think boolean dtype is specially handed in Arrow, but needs double check~)